runtime/weak.c

```diff
@@ -529,6 +574,23 @@ CAMLexport void caml_ephemeron_blit_key(value ars, mlsize_t offset_s,
   offset_s += CAML_EPHE_FIRST_KEY;
   offset_d += CAML_EPHE_FIRST_KEY;
 
+  if ( caml_gc_phase == Phase_mark
+       && ephe_list_pure
+       && Field(ard, CAML_EPHE_DATA_OFFSET) != caml_ephe_none ){
+
+    dest_has_white_value = 0;
+
+    for(i = 0; i < length; i++){
+      dest_has_white_value &= Is_White_During_Mark(Field(ard, offset_d + i));
```

Comment:
```suggestion
      dest_has_white_value |= Is_White_During_Mark(Field(ard, offset_d + i));
```
otherwise `dest_has_white_value` is always 0.